apps/code/editor_view.cpp

```diff
@@ -67,26 +81,27 @@ void EditorView::GutterView::drawRect(KDContext * ctx, KDRect rect) const {
   KDCoordinate firstLine = m_offset / glyphSize.height();
   KDCoordinate firstLinePixelOffset = m_offset - firstLine * glyphSize.height();
 
-  char lineNumber[k_lineNumberCharLength];
+  char lineNumber[m_digits];
   int numberOfLines = bounds().height() / glyphSize.height() + 1;
   for (int i=0; i<numberOfLines; i++) {
-    // Only the first two digits are displayed
-    int lineNumberValue = (i + firstLine + 1) % 100;
+    int lineNumberValue = (i + firstLine + 1);
     Poincare::Integer line(lineNumberValue);
-    if (firstLine < 10 || lineNumberValue >= 10) {
-      line.serialize(lineNumber, k_lineNumberCharLength);
-    } else {
-      // Add a leading "0"
-      lineNumber[0] = '0';
-      line.serialize(lineNumber + 1, k_lineNumberCharLength - 1);
+
+    int lineDigits = getDigits(lineNumberValue);
+
+    for (int j=0; j < (m_digits - lineDigits - 1); j++) {
+      lineNumber[j] = '0';
```

Comment:
Why did you use a 0 ?
```suggestion
      lineNumber[j] = ' ';
```